robot/impl/local_robot.go

```diff
@@ -667,7 +685,7 @@ func (r *localRobot) getDependencies(
 		// Specifically call ResourceByName and not directly to the manager since this
 		// will only return fully configured and available resources (not marked for removal
 		// and no last error).
-		r, err := r.ResourceByName(dep)
+		r, err := r.manager.ResourceByName(dep)
```

Comment:
[q] Why do we need this change, again? In any case, the comment right above it no longer seems accurate (we are directly calling `ResourceByName` on the manager).

Reply:
+1 to the question, are we correctly returning the right set of resources here (based on the comment above saying that resource manager may return resources marked for removal)?